### ISSUE TITLE
:rocket: Modernize Codebase

### ISSUE THREAD

**Issue by MBM1607** [jimniels/teamcolors#79]
Hi @jimniels,

I want to work on upgrading the repo all over. For this I want to do the following changes:

- [ ] :arrow_up: Upgrade to React 18 and Functional Components
- [ ] :safety_vest: Add TypeScript
- [ ] :building_construction: Swap out CRA for [Vite](https://vitejs.dev/)
- [ ] :package: Switch to use [PNPM](https://pnpm.io/) instead of yarn
- [ ] :art: Improve Site Theming
- [ ] :vampire: Add Dark Mode
- [ ] :hammer: Complete the [To-Do list](https://github.com/jimniels/teamcolors#to-dos)

I am creating this issue to indicate that I am interested in working on this. Please let me know if this is something that you would consider merging and if you have any concerns or questions.

I think this is a cool project and should be kept up to date.

**Comment by jimniels** [comment]
Thanks for your interest in the project :)

I don't know if I'm convinced this project needs react 18 + vite + pnpm + typescript. In hindsight, it had way too much tooling for what it is. I haven't revisited this project in a long time and now I can't even get it to deploy because the build won't work and I don't really have the interest/time to look into why at the moment.

The site itself is really just a static page. If I had to redo this, I would make it much simpler and just use vanilla HTML/CSS/JS with JSdoc annotations for any in-editor typing. In other words, my recommendation to make this up to date would be:

- Strip out React
- Use JSdoc annotations for any typing
- Get rid of CRA altogether
- Use npm (if necessary) for dev deps
- Make the site require no build step

That would make ongoing maintenance of this site into the future much easier. And it would decrease the amount of complexity in setting up, developing, and deploying the site (not to mention reduce the amount of code shipped to the client).

All that said, I'm not really convinced this project needs a technical facelift. What it needs is more accurate, up-to-date team information with the ability to accurately cite the source for its colors. Most of the original sourcing is no longer accessible.

In other words, this site has more of a content issue than technological one IMO.

Anyhow, I am hesitant to dump a bunch of new tech on something that was already over-tooled when I first made it. If there's any technological change, I'd prefer it be towards _simplification_ instead of _modernization_.

That said, if you want to work on it because you want to use those particular tools, you're always welcome to fork it and deploy/host it wherever you please :)

**Comment by muhammadk1607** [comment]
Thanks for the detailed reply. I get where you are coming from. 

I think I will go ahead with forking it and working on these changes in https://github.com/MBM1607/teamcolors.

I saw the source issue as well, for my changes when I have better sources, I will be sure to make a pull request to update only that part in the upstream as well.
